### ISSUE TITLE
404 on Tickets for Not Logged in members

### ISSUE THREAD

**Issue by wptech68** [nanodesigns/nanosupport#27]
When someone visit the ticket URL (permalink) without being loged in, it shows the standard 404 error page.
I think it should show something like: "You need to login first to view this ticket" so user knows why the ticket was not found, if not looks like it was removed from the system

**Comment by mayeenulislam** [comment]
That's the default behavior of WordPress Private posts.
Let's see and hope how can we change the behavior...

**Comment by wptech68** [comment]
ohh, I see
no problem, a 404 plugin will make the trick

**Comment by mayeenulislam** [comment]
After consulting many [possible] solutions, like:

1. [Prevent private post 404 - WPSE](https://wordpress.stackexchange.com/a/101981/22728)
2. [Show a special message for private page? - WPSE](https://wordpress.stackexchange.com/q/102903/22728)
3. ...

we actually came to an ending that it's a kind of restricted by WP Core.

Though we thought some ways, like grabbing the `slug` from the URL and find the post using a db query to identify a ticket and redirect users accordingly, using:

- [Wordpress query single post by slug - StackOverflow](https://stackoverflow.com/a/29770273/1743124)

But as WordPress permalink might not always stick with a single pattern, so identifying slug won't be easy. We're restricted by the WP Core by many means:

- [How to retrieve the slug of current page? - WPSE](https://wordpress.stackexchange.com/a/188945/22728)

So we concluded, it's a **Won't Fix** for NanoSupport as of _now_.
